algorithms/next_greater_number/C++/solution.cpp

```diff
@@ -0,0 +1,35 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+
+	char num[100000];
+
+	cin >> num;
+	if(!next_permutation(num, num+strlen(num)))
```

Comment:
@Aj163 Please implement the function yourself instead of using a library function.